src/coreclr/vm/assembly.cpp

```diff
@@ -52,6 +52,7 @@
 //#define STRICT_JITLOCK_ENTRY_LEAK_DETECTION
 //#define STRICT_CLSINITLOCK_ENTRY_LEAK_DETECTION
 
+LPCWSTR Assembly::s_wszDiagnosticStartupHookPaths = nullptr;
```

Comment:
Why not make this into a static variable local to this file. The only new member on the class would be the AddDiagnosticStartupHook. I would try to avoid adding private members to classes unless necessary, it has no real benefit and it makes compilation/maintenance harder as any changes have to touch two files (and touching the header trigger rebuilding of lot of dependencies).